sklearn/base.py

```diff
@@ -335,6 +337,30 @@ def _get_tags(self):
         tags.update(collected_tags)
         return tags
 
+    def _validate_n_features(self, X, check_n_features):
+        if check_n_features:
+            if not hasattr(self, 'n_features_in_'):
+                raise RuntimeError(
+                    "check_n_features is True but there is no n_features_in_ "
+                    "attribute."
+                )
+            if X.shape[1] != self.n_features_in_:
+                raise ValueError(
+                    'X has {} features, but this {} is expecting {} features '
+                    'as input.'.format(X.shape[1], self.__class__.__name__,
+                                       self.n_features_in_)
+                )
```

Comment:
should there be an "else" here? It would be equivalent but maybe make the intent more clear?